ckanext/recombinant/controller.py

```diff
@@ -181,7 +181,42 @@ def template(self, dataset_type, lang, owner_org):
         except NotFound:
             abort(404, _('Not found'))
 
+
         book = excel_template(dataset_type, org)
+
+
+        if request.method == 'POST':
+            filters = {}
+            resource_name = request.POST.get('resource_name','' )
+            for r in dataset['resources']:
+
+                if r['name'] == resource_name:
+                    resource = r
+                    break
+            else:
+                abort(400,"Resource not found")
+
+            pk_fields = recombinant_primary_key_fields(resource['name'])
+            key_indices = request.POST.get('key_indices').split(",")
+            primary_keys = []
+            for index in key_indices:
+                primary_keys.append(request.POST.get(index,'').split(","))
```

Comment:
request.POST is a MultiDict which has a `request.POST.getall('index')` that will return a list of all the values passed for the parameter "index". e.g. a form with multiple fields with repeated `<input name="index">` fields. Using this should reduce these 5 lines of code down to 1 here and simplify your JS too.